### PR TITLE
Add default ulimit test for gen kube

### PR DESCRIPTION
**Pull request by umohnani8**
Add a test for generate kube to verify that the ulimit
annotation is not set for the default case when the user
doesn't set any ulimits.

Fixes https://github.com/containers/podman/issues/18555
Implementation was done in https://github.com/containers/podman/pull/18721 and this PR adds a regression test.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: libpod/util.go [hunk 1]
```diff
@@ -369,3 +372,45 @@ func writeStringToPath(path, contents, mountLabel string, uid, gid int) error {
 
 	return nil
 }
+
+// getDefaultUlimits returns the default ulimits set based on whether podman is invoked
+// as rootless or root. This is used when determining what annotations to add for ulimits
+// during kube generate
+func getDefaultUlimits(limits []string) ([]string, error) {
```
Comment (Luap99) [inline libpod/util.go:379]: please add comments that this is the reverse implementation from addRlimits() and must be kept in sync

**Review by Luap99 — COMMENTED** [review]
I think the current way of setting default ulimits is just broken, see #18714 and #18696.
Instead of keeping this it is IMO better to move setting the default limits down in libpod and set it on each start. Then we do not alter ctr.config.Spec which means your kube generate code can just stay as it without changes.

File: libpod/util.go (hunk 1 above)
Comment (umohnani8) [reply libpod/util.go]: I moved the code around a bit so we are not duplicating implementation of grabbing the default ulimit list.

**Comment by Luap99** [comment]
@umohnani8 I think if we merge #18721 then you do not need any code changes at all, you could just add your test to prevent future regressions.

**Comment by umohnani8** [comment]
@containers/podman-maintainers PTAL

**Comment by baude** [comment]
im taking comments from @Luap99 to indicate his LGTM

/lgtm

**Pull request merged**